src/schema.c

```diff
@@ -313,3 +317,149 @@ ldap_schema_ready(struct ldap_connection_ctx_t* connection)
         return true;
     }
 }
+
+bool
+ldap_schema_validate_entry(ldap_schema_t* schema, ld_entry_t* entry, char** objectclass_names)
+{
+    return_null_if_null(schema, "ldap_schema_validate_entry - schema is NULL!\n");
+    return_null_if_null(entry, "ldap_schema_validate_entry - entry is NULL!\n");
+    return_null_if_null(entry, "ldap_schema_validate_entry - objectclass_names is NULL!\n");
+
+    LDAPAttribute_t* objectclass_attribute = ld_entry_get_attribute(entry, "objectClass");
+    return_null_if_null(objectclass_attribute, "ldap_schema_validate_entry - attribute objectClass is NULL!\n");
+
+    char** objectclass_attribute_values = objectclass_attribute->values;
+    return_null_if_null(objectclass_attribute_values, "ldap_schema_validate_entry - attribute objectClass values is NULL!\n");
+
+    LDAPObjectClass* objectclass;
+    for (int i = 0; objectclass == NULL && objectclass_names[i] != NULL; ++i)
```

Comment:
While technically `for` cycle allows using few variables in this case we should consider using while or moving `objectclass` variable out of cycle condition section.